### ISSUE TITLE
Create Blockchain Reference section

### ISSUE THREAD

**Issue by s-ben** [jholdstock/dcrdevdocs#27]
Right now, we have a number of pages in `Core Blockchain Concepts` that are more technical references than "core concepts" (e.g. `transaction format`, `addresses`, `utxo-set-semantics`). Should we break these out into another top-level menu? Like `Blockchain Reference` or similar? Bitcoin dev docs do something similar with their [Bitcoin Developer Reference](https://bitcoin.org/en/developer-reference) which is separate from their [Blockchain Guide](https://bitcoin.org/en/blockchain-guide) (which we're kind of using as a template).

**Comment by jholdstock** [comment]
Yeah I am totally happy for the menu structure to change. I had figured it was just a temporary thing to get us started anyway.
